firebase-dataconnect/firebase-dataconnect.gradle.kts

```diff
@@ -0,0 +1,234 @@
+  implementation(libs.grpc.protobuf.lite)
+  implementation(libs.grpc.kotlin.stub)
+  implementation(libs.grpc.stub)
+  implementation(libs.protobuf.java.lite)
+  implementation(libs.protobuf.kotlin.lite)
+
+  testCompileOnly(libs.protobuf.java)
+  testImplementation(project(":firebase-dataconnect:testutil"))
+  testImplementation(libs.androidx.test.junit)
+  testImplementation(libs.kotest.assertions)
+  testImplementation(libs.kotest.property)
+  testImplementation(libs.mockk)
+  testImplementation(libs.robolectric)
+  testImplementation(libs.truth)
+  testImplementation(libs.truth.liteproto.extension)
+  testImplementation(libs.kotlin.coroutines.test)
+  testImplementation(libs.kotlinx.serialization.json)
+
+  androidTestImplementation(project(":firebase-dataconnect:androidTestutil"))
+  androidTestImplementation(project(":firebase-dataconnect:connectors"))
+  androidTestImplementation(project(":firebase-dataconnect:testutil"))
+  androidTestImplementation("com.google.firebase:firebase-auth:22.3.1") {
+    exclude(group = "com.google.firebase", module = "firebase-common")
+    exclude(group = "com.google.firebase", module = "firebase-components")
+  }
+  androidTestImplementation("com.google.firebase:firebase-appcheck:18.0.0") {
+    exclude(group = "com.google.firebase", module = "firebase-common")
+    exclude(group = "com.google.firebase", module = "firebase-components")
+  }
+  androidTestImplementation(libs.androidx.test.core)
+  androidTestImplementation(libs.androidx.test.junit)
+  androidTestImplementation(libs.androidx.test.rules)
+  androidTestImplementation(libs.androidx.test.runner)
+  androidTestImplementation(libs.kotlin.coroutines.test)
+  androidTestImplementation(libs.kotest.assertions)
+  androidTestImplementation(libs.kotest.property)
+  androidTestImplementation(libs.mockk)
+  androidTestImplementation(libs.mockk.android)
+  androidTestImplementation(libs.truth)
+  androidTestImplementation(libs.truth.liteproto.extension)
+  androidTestImplementation(libs.turbine)
+}
+
+tasks.withType<KotlinCompile>().all {
+  kotlinOptions {
+    freeCompilerArgs = listOf("-opt-in=kotlin.RequiresOptIn")
+  }
+}
+
+extra["packageName"] = "com.google.firebase.dataconnect"
+
+tasks.withType<DokkaTask>().configureEach {
+  moduleName.set("firebase-dataconnect")
+  val cacheRootDirectory = layout.buildDirectory.dir("dokka/cache")
+  cacheRootDirectory.get().asFile.mkdirs()
+  cacheRoot.set(cacheRootDirectory)
+}
+
+// Enable Kotlin "Explicit API Mode". This causes the Kotlin compiler to fail if any
+// classes, methods, or properties have implicit `public` visibility. This check helps
+// avoid  accidentally leaking elements into the public API, requiring that any public
+// element be explicitly declared as `public`.
+// https://github.com/Kotlin/KEEP/blob/master/proposals/explicit-api-mode.md
+// https://chao2zhang.medium.com/explicit-api-mode-for-kotlin-on-android-b8264fdd76d1
+tasks.withType<KotlinCompile>().all {
```

Comment:
oo I didn't know this was a setting. tbh, this might be a good thing to enable globally for Kotlin SDKs. Possible future work (for investigation) @rlazo

Reply:
I've created [b/369383096](https://b.corp.google.com/issues/369383096) to track this.